### PR TITLE
Ignore scroll wheel with Ctrl pressed

### LINKED ISSUE #28: Support Google maps
Google maps uses a ctrl+scrollwheel event to zoom in and out, without hijacking the standard scroll event. 

If we ignore scroll wheel when ctrl is pressed, this will prevent both the horizontal scroll and map handling the scroll. It will also allow the ctrl+wheel to control zoom in Chrome.

### PR DESCRIPTION
**Pull request by Mark-Hetherington**
Fixes #28

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -151,6 +151,8 @@ export default class HorizontalScroll extends EventEmitter {
     }
 
     private wheel = (e: WheelEvent) => {
+        if (e.ctrlKey) // Ignore scroll event if ctrl key pressed
```
Comment (matthijsoberon) [inline src/index.ts:154]: I'd suggest to make it more generic by passing a function, in which it is determined if the scroll should be ignored, to the constructor. This way we make sure to make the ignoring of the scroll event configurable.
Example: 
``` 
useHorizontal({ ignoreScroll: (e) => {
    return e.ctrlKey;
}});
```

**Review by matthijsoberon — CHANGES_REQUESTED** [review]

**Comment by matthijsoberon** [comment]
Nice PR, I've put some feedback on it to make it a little more generic. Thanks for contributing 👍

File: package.json [hunk 2]
```diff
@@ -9,7 +9,8 @@
         "build": "npm run clean && run-s build:*",
         "build:lib": "tsc --project .",
         "build:toc": "doctoc --notitle README.md",
-        "clean": "rimraf *.js *.d.ts *.js.*"
+        "clean": "rimraf *.js *.d.ts *.js.*",
+        "prepare": "npm run build"
```
[the remark this answers is not in the document]
Comment (Mark-Hetherington) [reply package.json:13]: Ok, sure. 

Outside the scope of this specific pull request, I'm not familiar enough with the general workflow of node/npm via git to know if this is a good or bad idea. It seems to be required to be able to work on this from a fork. But perhaps there is a better workflow?

File: src/index.ts (hunk 1 above)
Comment (Mark-Hetherington) [reply src/index.ts]: I've made an attempt at creating a Callback interface, and using that in the options structure.

File: src/index.ts [hunk 3]
```diff
@@ -46,6 +51,7 @@ export default class HorizontalScroll extends EventEmitter {
         container = document.documentElement,
         showScrollbars = false,
         preventVerticalScroll = false,
+        ignoreScroll = IGNORE_SCROLL
```
Comment (matthijsoberon) [inline src/index.ts:54]: Remove the default ignore_scroll behaviour, as this is used for specific use cases.

**Comment by matthijsoberon** [comment]
Gave some other feedback aswell. Looks great! Just one question; should the "ignoreScroll" just ignore the scrolling vertically? Because that's what's happening right now with the implementation.

**Comment by matthijsoberon** [comment]
Took over this PR and made a new PR #33 with some minor changes

**Pull request closed without merging**